### ISSUE TITLE
Non-reactive TokenValidator

### ISSUE THREAD

**Issue by Spikhalskiy** [micronaut-projects/micronaut-security#1602]
### Feature description

In recent versions AuthenticationProvider became non-reactive and the reactive version was deprecated.
It would be great to see TokenValidator to become non-reactive too or get an alternative non-reactive version.
Currently, it's not clear if a correct TokenValidator implementation can be blocking. And if it can, why TokenValidator have to return a Publisher.

**Comment by sdelamo** [comment]
I agree with you @Spikhalskiy we should follow a similar strategy providing imperative and reactive APIs. 

`TokenValidator` API is reactive because a user may want to do a blocking operation in a token validator. E.g. do a blocking call to the database to verify the API token supplied is valid.

**Comment by Spikhalskiy** [comment]
@sdelamo yeah, I understand that such scenario is the case for the original thinking behind returning a Publisher. What is not clear to me from the docs at least, is it ok to put a blocking operation inside `TokenValidator#validateToken` call or this call should return "immediately" with a publisher that can be filled later with the validation result?

If it's ok to place the blocking code there, in that case returning a publisher is only confusing and not really helpful. The only situation when anyone will use a delayed publishing is if they are worried about an amount of waiting threads.

If it's not ok, well, this definitely should be in the javadocs and we need to fix our implementation :)

**Comment by sdelamo** [comment]
It is not ok to put blocking code. I think improving the javadoc is a good idea. if you are interesting in contributing such an improvement let me know.
